### PR TITLE
feat(docs): i18n stuff

### PR DESCRIPTION
**Pull request by c-ehrlich**
Resolves https://github.com/t3-oss/create-t3-app/issues/821, https://github.com/t3-oss/create-t3-app/issues/792

## ✅ Checklist

- [x] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-10-06).
- [x] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I performed a functional test on my final commit

---

## Changelog

Some more preparations for getting translations online

- show table of contents for active language
- sidebar can be translated
- a lot of styling fixes for rtl languages

Will require more work in the future, but i think it makes sense to merge WIP occasionally

---

## Screenshots

image

💯

### CODE REVIEW

**Comment by juliusmarminge** [comment]
Could have just continued on my PR :(

**Comment by c-ehrlich** [comment]
> Could have just continued on my PR :(

oh yeah my bad :/

feel free to merge this into your pr and close

**Comment by Yevhenii-Bakhmat** [comment]
I think parsing a titles from tanslation pages would be better then using constant with paths and labels

**Comment by c-ehrlich** [comment]
> I think parsing a titles from tanslation pages would be better then using constant with paths and labels

i agree this would be better, but can't think of an easy way to do it currently. only way i can think of is to import every article in each language in config.ts and get frontmatter from all of them, which is a lot. maybe there's something to help us in astro docs, can check tonight.

File: www/src/components/navigation/leftSidebar.astro [hunk 1]
```diff
@@ -50,14 +57,14 @@ const sidebar = SIDEBAR[langCode];
...
-                          "border-t3-purple-300 bg-t3-purple-300/30 font-medium dark:bg-t3-purple-300/20":
-                            isActive,
-                          "border-t3-purple-300/20": !isActive,
-                        },
-                      )}
+                      class={clsx([
+                        "text-md text-t3-purple-800 hover:border-t3-purple-300/50 hover:bg-t3-purple-300/20 hover:text-t3-purple-400 dark:text-t3-purple-200 dark:hover:bg-t3-purple-300/10 dark:hover:text-t3-purple-100 block py-2 transition-colors",
+                        isActive &&
+                          "border-t3-purple-300 bg-t3-purple-300/30 dark:bg-t3-purple-300/20 font-medium",
+                        isActive && "border-t3-purple-300/20",
```
Comment (juliusmarminge) [inline www/src/components/navigation/leftSidebar.astro:64]: prob missed an `!` here? both entries are `isActive` now

Comment (c-ehrlich) [reply www/src/components/navigation/leftSidebar.astro]: yup, fixed now

**Comment by juliusmarminge** [comment]
What's the status here?

**Comment by c-ehrlich** [comment]
> What's the status here?

Need to resolve conflicts tomorrow, other than that ready to review/merge.

**Comment by c-ehrlich** [comment]
ready for review now @juliusmarminge

**Comment by juliusmarminge** [comment]
too disappears https://share.cleanshot.com/502Vvd

**Comment by juliusmarminge** [comment]
This banner also looks a bit weird, but honestly im not sure how it should look so can't say 
CleanShot 2022-11-27 at 13 23 57@2x

**Comment by c-ehrlich** [comment]
> This banner also looks a bit weird, but honestly im not sure how it should look so can't say CleanShot 2022-11-27 at 13 23 57@2x

There are two options for this
1. Always display it in English, then it should be ltr
2. Translate this banner for other languages (seems to make sense), then it should be rtl as it is now.
if we decide to start translating not just .md files, we'll probably want https://www.npmjs.com/package/astro-i18next

Re: ToC - will check what's going on tomorrow

**Comment by juliusmarminge** [comment]
> Always display it in English, then it should be ltr

This for now maybe? Can always change later

**Comment by c-ehrlich** [comment]
image
something weird going on with the external link icons in rtl

**Comment by c-ehrlich** [comment]
image

**Comment by c-ehrlich** [comment]
also dont merge before arabic is removed from the list of known languages, otherwise we're deploying two fake arabic pages

File: cli/package.json [hunk 2]
```diff
@@ -54,8 +54,8 @@
     "@types/gradient-string": "^1.1.2",
     "@types/inquirer": "^9.0.2",
     "@types/node": "^18.8.0",
-    "prettier": "^2.7.1",
-    "prettier-plugin-tailwindcss": "0.2.0-beta.3",
+    "prettier": "^2.8.0",
+    "prettier-plugin-tailwindcss": "^0.2.0",
```
Comment (juliusmarminge) [inline cli/package.json:58]: i wanted to use satisfies but didn't work.

upgraded these to get the parsing correct

**Comment by c-ehrlich** [comment]
@julius lmk when you're ready to merge and i'll remove the arabic placeholder stuff (or feel free to remove it yourself)

File: .changeset/heavy-buckets-try.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"create-t3-app": patch
+---
+
+upgrade prettier and eslint (for docs)
```
Comment (juliusmarminge) [inline .changeset/heavy-buckets-try.md:5]: rm this.

we're not updating any cli logic or files for the scaffolded app, so a publish here won't do anything. it's just devDeps being updated

**Review by juliusmarminge — CHANGES_REQUESTED** [review]

**Review by juliusmarminge — APPROVED** [review]
cool lets ship

**Pull request merged**